packages/runtime/src/interpreter/opcode-list.ts

```diff
@@ -1793,9 +1793,9 @@ export class AppLocalPut extends Op {
     this.assertMinStackLen(stack, 3, this.line);
     const value = stack.pop();
     const key = this.assertBytes(stack.pop(), this.line);
-    const accountIndex = this.assertBigInt(stack.pop(), this.line);
+    const accountRef: StackElem = stack.pop();
```

Comment:
Here we should follow different approach. what if version here is 3 ? the same opcode will behave differently.
so maybe we can use opcodev2, opcodev4 as suggested by robert.

Reply:
I think it's better to resolve an assetId/appId by reference in this case (opcode & function being teal version agnostic). Previously it was confusing resolving these id's in separate ops (we were handling lot of cases). So i prefer to keep this in a single function [https://github.com/algorand/go-algorand/blob/master/data/transactions/logic/eval.go#L3041](https://github.com/algorand/go-algorand/blob/master/data/transactions/logic/eval.go#L3041)

Reply:
Is it possible that this opcode will behave differently in different versions ?
Or am I misunderstanding this here ?